pkg/build/api/validation/validation.go

```diff
@@ -683,7 +683,12 @@ func ValidateStrategyEnv(vars []kapi.EnvVar, fldPath *field.Path) field.ErrorLis
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), ev.Name, cIdentifierErrorMsg))
 		}
 		if ev.ValueFrom != nil {
-			allErrs = append(allErrs, field.Invalid(idxPath.Child("valueFrom"), ev.ValueFrom, "valueFrom is not supported in build strategy environment variables"))
+			if ev.Value != "" {
+				allErrs = append(allErrs, field.Duplicate(idxPath.Child("value"), fmt.Sprintf("cannot use 'value' and 'valueFrom' together for %v", ev.Name)))
+			}
+			if ev.ValueFrom.ResourceFieldRef != nil {
```

Comment:
I think that we can/should remove this code in favour of ValidateEnv() from vendor/k8s.io/kubernetes/pkg/api/validation, with an additional check if we don't permit `ResourceFieldRef != nil`.